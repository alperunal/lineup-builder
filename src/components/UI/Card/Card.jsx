import React from 'react';
import cn from 'classnames';

import './Card.scss';

const Card = ({
    title, description, customClass, imageSrc, imageAlt,
}) => (
    <div
        className={
            cn(
                customClass,
                'card',
            )
        }
    >
        {imageSrc ? (
            <img
                src={imageSrc}
                alt={imageAlt}
                className="card__image"
            />
        ) : null}
        <div className="card__info">
            <div className="card__title">
                {title}
            </div>
            <p className="card__description">
                {description}
            </p>
        </div>
    </div>
);

Card.defaultProps = {
    customClass: '',
    imageSrc: '',
    imageAlt: '',
    title: '',
    description: '',
};

export default Card;
