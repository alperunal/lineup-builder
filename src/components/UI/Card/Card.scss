.card {
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),
    0 4px 6px -2px rgba(0, 0, 0, 0.05);
  box-sizing: border-box;
  border: 0;
  border-radius: 0.25rem;

  &__image {
    max-width: 100%;
    height: auto;
  }

  &__info {
    padding-left: 1.5rem;
    padding-right: 1.5rem;
    padding-top: 1rem;
    padding-bottom: 1rem;
    box-sizing: border-box;
    border: 0;
  }

  &__title {
    margin-bottom: 0.5rem;
    font-size: 1.25rem;
    font-weight: 700;
    color: var(--color-text);
  }

  &__description {
    color: var(--color-text-secondary);
  }
}
