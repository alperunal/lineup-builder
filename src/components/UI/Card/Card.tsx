import React from 'react';
import classnames from 'classnames';

interface IProps {
    title?: string;
    type?: string;
    customClass?: string;
    children: React.ReactNode;
}

const Card: React.FC<IProps> = ({
    title, customClass, children, type,
}: IProps) => (
    <div
        className={classnames(customClass, 'card', {
            'card--success': type === 'success',
            'card--warning': type === 'warning',
            'card--danger': type === 'danger',
        })}
    >
        {title ? <div className="card__title">{title}</div> : null}
        {children}
    </div>
);

export default Card;
