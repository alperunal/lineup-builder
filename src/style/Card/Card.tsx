import React from 'react';
import classnames from 'classnames';

interface IProps {
    title?: string;
    type?: string;
    customClass?: string;
    children: React.ReactNode;
}

function Card({ title, customClass, children, type }: IProps) {
    return (
        <div
            className={classnames(customClass, 'card', {
                'card--success': type === 'success',
                'card--warning': type === 'warning',
                'card--danger': type === 'danger',
            })}
        >
            {title ? <div className="card__title">{title}</div> : null}
            {children}
        </div>
    );
}

export default Card;