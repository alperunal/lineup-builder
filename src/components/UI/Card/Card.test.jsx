import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

it('should render Card correctly', () => {
  const wrapper = shallow(
    <Card title="Test Title" description="Test Description" />,
  );

  expect(wrapper).toMatchSnapshot();
});
