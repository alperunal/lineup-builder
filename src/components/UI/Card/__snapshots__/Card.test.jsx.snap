// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Card correctly 1`] = `ShallowWrapper {}`;
